packages/jest-transform/src/__tests__/ScriptTransformer.test.ts

```diff
@@ -1816,6 +1822,19 @@ describe('ScriptTransformer', () => {
       ['test_preprocessor', expect.any(Object)],
     ]);
   });
+
+  it('regardless of sync/async, handles race condition where cache was deleted (fs.existsSync returns true but readFileSync returns ENOENT)', async () => {
+    fs.existsSync = jest.fn(() => true);
```

Comment:
Hi, can someone help me with this? I don't know how to properly override `fs` without affecting other test cases.

Reply:
`fs.existsSync.mockImplementaionOnce` perhaps? To only override it once and not for all invocations?

Reply:
@SimenB Thank you for your suggestion. You mean like this?

```diff
- fs.existsSync = jest.fn(() => true);
+ fs.existsSync.mockImplementationOnce(() => true);
```

I run it and seems like it has the same problem as my initial version. The main problem I have is whatever I mock here is not injected into the `createScriptTransformer` function. `existsSync` will still check with `mockFs` following its initial mock implementation. Even if I add `createScriptTransformer = require('../ScriptTransformer').createScriptTransformer;` after this override, the implementation won't update and still use the initial `fs` mock.

Reply:
Looking at the code, it seems we guard with `existsSync` in the `transform(Async)Source` case, not `transform(Sync)`. For `transform` we attempt to read first: https://github.com/facebook/jest/blob/b9fc8ac78ca6b6dfdb90797523c21d06e819512c/packages/jest-transform/src/ScriptTransformer.ts#L628 (before later calling `transformSource`)